packages/common/infra/src/livedata/index.ts

```diff
@@ -268,6 +269,38 @@ export class LiveData<T = unknown> implements InteropObservable<T> {
     this.upstreamSubscription?.unsubscribe();
   }
 
+  /**
+   * flatten the livedata
+   *
+   * ```
+   * new LiveData(new LiveData(0)).flat() // LiveData<number>
+   * ```
+   *
+   * ```
+   * new LiveData([new LiveData(0)]).flat() // LiveData<number[]>
+   * ```
+   */
+  flat(): T extends LiveData<infer P>
```

Comment:
should it be auto flatten like promise?

Reply:
I tried to implement auto flatten, and it works, but some cases are not very clear. You can still define `LiveData<LiveData<T>>` just like `Promise<Promise<T>>` which is legal in typescript but does not actually exist.